Indicators/HeikinAshi.cs

```diff
@@ -0,0 +1,117 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// This indicator computes the Heikin-Ashi bar (HA)
+    /// The Heikin-Ashi bar is calculated using the following formulas:
+    /// HA_Close[0] = (Open[0] + High[0] + Low[0] + Close[0]) / 4
+    /// HA_Open[0] = (HA_Open[1] + HA_Close[1]) / 2
+    /// HA_High[0] = MAX(High[0], HA_Open[0], HA_Close[0])
+    /// HA_Low[0] = MIN(Low[0], HA_Open[0], HA_Close[0])
+    /// </summary>
+    public class HeikinAshi : TradeBarIndicator
+    {
+        /// <summary>
+        /// Gets the Heikin-Ashi Open
+        /// </summary>
+        public IndicatorBase<IndicatorDataPoint> Open { get; private set; }
+
+        /// <summary>
+        /// Gets the Heikin-Ashi High
+        /// </summary>
+        public IndicatorBase<IndicatorDataPoint> High { get; private set; }
+
+        /// <summary>
+        /// Gets the Heikin-Ashi Low
+        /// </summary>
+        public IndicatorBase<IndicatorDataPoint> Low { get; private set; }
+
+        /// <summary>
+        /// Gets the Heikin-Ashi Close
+        /// </summary>
+        public IndicatorBase<IndicatorDataPoint> Close { get; private set; }
+
```

Comment:
I don't disagree with your use case, but what if I want to maintain a `RollingWindow<TradeBar>` that is the history of the indicator, or pass the trade bar around to other parts of the system which don't know about the `HeikenAshi` indicator.

Sounds good, thanks!